velox/functions/lib/Re2Functions.h

```diff
@@ -97,9 +101,14 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> re2ExtractSignatures();
 
 /// Return the pair {pattern kind, length of the fixed pattern} for fixed,
 /// prefix, and suffix patterns. Return the pair {pattern kind, number of '_'
-/// characters} for patterns with wildcard characters only. Return
-/// {kGenericPattern, 0} for generic patterns).
-std::pair<PatternKind, vector_size_t> determinePatternKind(StringView pattern);
+/// characters} for patterns with wildcard characters only. Return the pair
```

Comment:
We should not overload the return values with different meanings. Its better to create a struct with fields for the different outputs are set them correctly. You are also sending an output parameter to the function, so overall this is a bit confusing.
Its also better to write with a bulleted list
For prefix patterns : {function does...}
For suffix patterns : same as above
For wildcard patterns : ...